src/core/server/logging/logging_system.ts

```diff
@@ -204,18 +200,32 @@ export class LoggingSystem implements LoggerFactory {
       }
     }
 
-    for (const [loggerKey, loggerAdapter] of this.loggers) {
-      loggerAdapter.updateLogger(this.createLogger(loggerKey, computedConfig));
-    }
-
+    this.enforceConfiguredAppendersUsage(computedConfig);
     // We keep a reference to the base config so we can properly extend it
     // on each config change.
     this.baseConfig = newBaseConfig;
-    this.computedConfig = computedConfig;
 
     // Re-log all buffered log records with newly configured appenders.
     for (const logRecord of this.bufferAppender.flush()) {
       this.get(logRecord.context).log(logRecord);
     }
   }
+
+  // reconfigure all the loggers to have them use the buffer appender
+  // while we are awaiting for the appenders to be disposed.
+  private enforceBufferAppendersUsage() {
+    for (const [loggerKey, loggerAdapter] of this.loggers) {
+      loggerAdapter.updateLogger(this.createLogger(loggerKey, undefined));
+    }
+
+    // new loggers created during applyBaseConfig execution should use the buffer appender as well
+    this.computedConfig = undefined;
+  }
+
+  private enforceConfiguredAppendersUsage(config: LoggingConfig) {
+    for (const [loggerKey, loggerAdapter] of this.loggers) {
+      loggerAdapter.updateLogger(this.createLogger(loggerKey, config));
+    }
+    this.computedConfig = config;
```

Comment:
NIT: I wonder if this logic could be extracted to a common method where `config` can be `undefined`. Then, `applyBaseConfig` has the _business logic_ that decides when to call it with `undefined` and when with a provisioned config. Does it make sense?

Reply:
Maybe, but I find both method names to be more self-descriptive. Also, we might need different logic in the future, so not sure we should strive for DRY in this case.